arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift

```diff
@@ -710,3 +748,37 @@ extension AGSLoadStatus {
         }
     }
 }
+
+extension String {
+    func autoPanModeFromString() -> AGSLocationDisplayAutoPanMode? {
+        switch self {
+            case "compassNavigation":
+                return .compassNavigation
+            case "navigation":
+                return .navigation
+            case "recenter":
+                return .recenter
+            case "off":
+                return .off
+            default:
+                return nil
+        }
+    }
+}
+
+extension AGSLocationDisplayAutoPanMode {
+    func toName() -> String {
+        switch self {
+        case .off:
+            return "off"
+        case .recenter:
+            return "recenter"
+        case .navigation:
+            return "navigation"
+        case .compassNavigation:
+            return "compassNavigation"
+        @unknown default:
+            return "unknown"
```

Comment:
lets default to `nil` and fallback like you did in kotlin. I really liked that :D